### PR TITLE
subscription delivery queue configurable name

### PR DESCRIPTION
**Pull request by fil512**

### CODE REVIEW

File: hapi-fhir-jpaserver-subscription/src/main/java/ca/uhn/fhir/jpa/subscription/module/cache/SubscriptionRegistry.java [hunk 1]
```diff
@@ -48,16 +50,14 @@
...
 @Component
 public class SubscriptionRegistry {
-	private static final org.slf4j.Logger ourLog = org.slf4j.LoggerFactory.getLogger(SubscriptionRegistry.class);
+	private static final Logger ourLog = LoggerFactory.getLogger(SubscriptionRegistry.class);
 	private final ActiveSubscriptionCache myActiveSubscriptionCache = new ActiveSubscriptionCache();
 	@Autowired
 	SubscriptionCanonicalizer<IBaseResource> mySubscriptionCanonicalizer;
 	@Autowired
-	SubscriptionDeliveryHandlerFactory mySubscriptionDeliveryHandlerFactory;
+	ISubscriptionDeliveryChannelNamer mySubscriptionDeliveryChannelNamer;
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-subscription/src/main/java/ca/uhn/fhir/jpa/subscription/module/cache/SubscriptionRegistry.java]: Are these fields deliberately package-scoped for any reason?

File: hapi-fhir-jpaserver-subscription/src/main/java/ca/uhn/fhir/jpa/subscription/module/channel/SubscriptionChannelRegistry.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	private final Multimap<String, String> myActiveSubscriptionByChannelName = MultimapBuilder.hashKeys().arrayListValues().build();
+
+	@Autowired
+	SubscriptionDeliveryHandlerFactory mySubscriptionDeliveryHandlerFactory;
+	@Autowired
+	SubscriptionChannelFactory mySubscriptionDeliveryChannelFactory;
+	@Autowired
+	ModelConfig myModelConfig;
+
+	public void add(ActiveSubscription theActiveSubscription) {
```
Comment (jamesagnew) [inline hapi-fhir-jpaserver-subscription/src/main/java/ca/uhn/fhir/jpa/subscription/module/channel/SubscriptionChannelRegistry.java]: Should the add/remove methods be syncrhonized given their work against a shared map?

Comment (jamesagnew) [reply hapi-fhir-jpaserver-subscription/src/main/java/ca/uhn/fhir/jpa/subscription/module/channel/SubscriptionChannelRegistry.java]: In fact maybe every method here should be?

File: hapi-fhir-jpaserver-subscription/src/main/java/ca/uhn/fhir/jpa/subscription/module/cache/SubscriptionRegistry.java (hunk 1 above)
Comment (fil512) [reply hapi-fhir-jpaserver-subscription/src/main/java/ca/uhn/fhir/jpa/subscription/module/cache/SubscriptionRegistry.java]: Fixed.  Bad habit.

File: hapi-fhir-jpaserver-subscription/src/main/java/ca/uhn/fhir/jpa/subscription/module/channel/SubscriptionChannelRegistry.java (hunk 2 above)
Comment (fil512) [reply hapi-fhir-jpaserver-subscription/src/main/java/ca/uhn/fhir/jpa/subscription/module/channel/SubscriptionChannelRegistry.java]: Fixed.  Good catch.

**Pull request merged**